src/propagate-tor-test_cuda.cu

```diff
@@ -918,39 +924,73 @@ int main (int argc, char* argv[]) {
 
    printf("done preparing!\n");
  
-   printf("Number of struct MPTRK trk[] = %ld\n", nevts*nb);
-   printf("Number of struct MPTRK outtrk[] = %ld\n", nevts*nb);
-   printf("Number of struct struct MPHIT hit[] = %ld\n", nevts*nb);
+   printf("Number of struct MPTRK trk[] = %d\n", nevts*nb);
+   printf("Number of struct MPTRK outtrk[] = %d\n", nevts*nb);
+   printf("Number of struct struct MPHIT hit[] = %d\n", nevts*nb);
   
    printf("Size of struct MPTRK trk[] = %ld\n", nevts*nb*sizeof(struct MPTRK));
    printf("Size of struct MPTRK outtrk[] = %ld\n", nevts*nb*sizeof(struct MPTRK));
    printf("Size of struct struct MPHIT hit[] = %ld\n", nlayer*nevts*nb*sizeof(struct MPHIT));
 
-   
+   auto doWork = [&](const char* msg, int nIters) {
+     double wall_time = 0;
+
+#ifdef MEASURE_H2D_TRANSFER
+     for(int itr=0; itr<nIters; itr++) {
+       auto wall_start = std::chrono::high_resolution_clock::now();
+       for (int s = 0; s<num_streams;s++) {
+         transferAsyncTrk(trk_dev, trk,streams[s]);
+         transferAsyncHit(hit_dev, hit,streams[s]);
+       }
```

Comment:
Should there be a `cudaDeviceSynchronize()` here?

Reply:
A `cudaDeviceSynchronize()` is not needed here because the kernels and transfers to host are queued in the same CUDA stream.